### PR TITLE
Added LRN, MaxPool with Strides, ZeroPadding2D

### PR DESCRIPTION
**Pull request by pranv**

### CODE REVIEW

File: keras/layers/recurrent.py [hunk 1]
```diff
@@ -300,19 +292,15 @@ class LSTM(Recurrent):
...
-
         and returns outputs with shape:
         if not return_sequences:
             (nb_samples, output_dim)
         if return_sequences:
             (nb_samples, max_sample_length, output_dim)
-
         For a step-by-step description of the algorithm, see:
         http://deeplearning.net/tutorial/lstm.html
-
```
Comment (fchollet) [inline keras/layers/recurrent.py]: I'm fine with removing end-of-line extra spaces, but the spacing in the comments is here for a reason (readability). Can you exclude from this PR the files that are outside of the scope of the PR?

**Comment by pranv** [comment]
It was stupid Sublime Text that removes trailing whitespaces in each line. I f**k up at too many places whenever I want to contribute something :(

**Comment by fchollet** [comment]
There are just too many commits in this PR (which will make the git history hard to browse) and too many changes unrelated to the PR. 

Here's what you should do: 
- start from fchollet/master.
- add LRN2D, MaxPooling with strides, ZeroPadding.
- commit (you can do this in 3 commits, one per feature. One commit is fine too.), and create a PR.
